flink-streaming-connectors/flink-connector-hbase/src/main/java/org/apache/flink/streaming/connectors/hbase/MutationActionList.java

```diff
@@ -0,0 +1,371 @@
+
+import org.apache.flink.util.Preconditions;
+import org.apache.hadoop.hbase.client.Append;
+import org.apache.hadoop.hbase.client.Delete;
+import org.apache.hadoop.hbase.client.Durability;
+import org.apache.hadoop.hbase.client.Increment;
+import org.apache.hadoop.hbase.client.Mutation;
+import org.apache.hadoop.hbase.client.Put;
+
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ *  This class represents a list of {@link MutationAction}s you will take when writing
+ *  an input value of {@link HBaseSink} to a row in a HBase table.
+ *  Each {@link MutationAction} can create an HBase {@link Mutation} operation type
+ *  including {@link Put}, {@link Increment}, {@link Append} and {@link Delete}.
+ */
+public class MutationActionList {
+	private final List<MutationAction> actions;
+
+	public MutationActionList() {
+		this.actions = new ArrayList<>();
+	}
+
+	public List<MutationAction> getActions() {
+		return this.actions;
+	}
+
+	/**
+	 * Create a new list of HBase {@link Mutation}s.
+	 *
+	 * @param rowKey row that the created {@link Mutation} list is applied to
+	 * @param writeToWAL enable WAL
+	 * @return a list of HBase {@link Mutation}s
+	 */
+	public List<Mutation> newMutationList(byte[] rowKey, boolean writeToWAL) {
+		List<Mutation> mutations = new ArrayList<>();
+		Put put = null;
+		Increment increment = null;
+		Append append = null;
+		Delete delete = null;
+		boolean rowIsDeleted = false;
+		for (MutationAction action : actions) {
+			switch (action.getType()) {
+				case PUT:
+					if (put == null) {
+						put = new Put(rowKey);
+						mutations.add(put);
+					}
+					if (action.getTs() == -1) {
+						put.addColumn(action.getFamily(), action.getQualifier(), action.getValue());
+					} else {
+						put.addColumn(action.getFamily(), action.getQualifier(), action.getTs(), action.getValue());
+					}
+					break;
+
+				case INCREMENT:
+					if (increment == null) {
+						increment = new Increment(rowKey);
+						mutations.add(increment);
+					}
+					increment.addColumn(action.getFamily(), action.getQualifier(), action.getIncrement());
+					break;
+
+				case APPEND:
+					if (append == null) {
+						append = new Append(rowKey);
+						mutations.add(append);
+					}
+					append.add(action.getFamily(), action.getQualifier(), action.getValue());
+					break;
+
+				// If there are multiple DELETE_ROW actions, only the first one is served
+				case DELETE_ROW:
+					if (!rowIsDeleted) {
+						for (int i = 0; i < mutations.size(); ) {
+							if (mutations.get(i) instanceof Delete) {
+								mutations.remove(i);
+							} else {
+								i++;
+							}
+						}
+						delete = new Delete(rowKey, action.getTs());
+						mutations.add(delete);
+						rowIsDeleted = true;
+					}
+					break;
+
+				case DELETE_FAMILY:
+					if (!rowIsDeleted) {
+						if (delete == null) {
+							delete = new Delete(rowKey);
+							mutations.add(delete);
+						}
+						delete.addFamily(action.getFamily(), action.getTs());
```

Comment:
This case is already covered when calling MutationActionList#addDeleteFamily without timestamp, then ts field is assigned with Long.MAX_VALUE which will cause all columns with the specified family to be deleted.